src/libs/HttpUtils.js

```diff
@@ -109,10 +108,13 @@ function xhr(command, data, type = CONST.NETWORK.METHOD.POST, shouldUseSecure =
 
     let apiRoot = shouldUseSecure ? CONFIG.EXPENSIFY.SECURE_EXPENSIFY_URL : CONFIG.EXPENSIFY.URL_API_ROOT;
 
-    if (shouldUseStagingServer(stagingServerToggleState)) {
-        apiRoot = shouldUseSecure ? CONFIG.EXPENSIFY.STAGING_SECURE_EXPENSIFY_URL : CONFIG.EXPENSIFY.STAGING_EXPENSIFY_URL;
+    if (shouldUseStagingServer) {
+        if (CONFIG.IS_USING_WEB_PROXY) {
+            apiRoot += shouldUseSecure ? CONST.API_MAP.STAGING_SECURE : CONST.API_MAP.STAGING;
```

Comment:
I'm not sure whether we should build the URL in this `+=` matter.
Mainly because it's harder to trace the complete value like we can we the rest of the urls from CONST
Also it makes it necessary to introduce `CONST.API_MAP` which only has the staging values and is only used here if `IS_USING_WEB_PROXY` is `true` - `API_MAP` is too generic term for such a specific case
The least it needs a lot of comments explaining all these changes

---

Since this is DEV related it might be best to keep `CONST.API_MAP` out of `CONST.js` and have it in a file that both `web/proxy.js` and `libs/HttpUtils.js` can access 
_(perhaps a `config/proxyConfig.js` file)_
This serve 2 things 
1. Makes it clear it's only about the PROXY / DEV
2. Code reuse - we declare PROXY_STAGING_HOST and PROXY_STAGING_SECURE_HOST in one place